### ISSUE TITLE
Improvements to the summary sheet

### ISSUE THREAD

**Issue by miketaylr** [mozilla/tsci#93]
I'd like to fix a few things:

* [x] If you create a new spreadsheet, there is no Summary sheet. We can fix that.
* [x] For said Summary, we should add headers.
* [x] Sheet updates stack in reverse chronological order (it would be nice if newer sheets would be added to the bottom row, not the first (second) row)
* [x] Possibly I screwed up what data gets sent to the Summary sheet -- fix anything that's broken.

**Comment by past** [comment]
>     * [ ]  Sheet updates stack in reverse chronological order (it would be nice if newer sheets would be added to the bottom row, not the first (second) row)

For context, I did this so that it would be easier to update existing spreadsheets. Changing the order will require parsing the existing summary sheet values and determining the line to append to. I think we can continue to ignore the sorting of the dates when updating, since it doesn't affect the graph.

**Comment by miketaylr** [comment]
> For context, I did this so that it would be easier to update existing spreadsheets. Changing the order will require parsing the existing summary sheet values and determining the line to append to.

The way I implemented it in #94 is to always append to the last row -- which is what [this doc does](https://docs.google.com/spreadsheets/d/1rmH1EhTVZBtTIXXwqSHcaQXhmv7xJh2Y2LuNiySPURQ/edit?ts=5cf460c1#gid=1949428969). If we pass this spreadsheet ID to a script, wouldn't it break the graph by placing 7/21 in the second row (rather than the last)? 

(Note: my use of "reverse chronological order" I think makes no sense, it should just be "chronological order"... or oldest row up top, moving toward the future as it goes down)

Or maybe I don't understand how the graph works...?

> I think we can continue to ignore the sorting of the dates when updating, since it doesn't affect the graph.

The way the script works on master is to produce something like this, which I think is the opposite of what we want: https://docs.google.com/spreadsheets/d/1abgFrbdp9c_9u5gItnuYCHoiSG-it-DpIPLJuty9erA/edit?ts=5cf460c1#gid=234433070

@past am i totally confused here?

**Comment by past** [comment]
> The way I implemented it in #94 is to always append to the last row -- which is what [this doc does](https://docs.google.com/spreadsheets/d/1rmH1EhTVZBtTIXXwqSHcaQXhmv7xJh2Y2LuNiySPURQ/edit?ts=5cf460c1#gid=1949428969). If we pass this spreadsheet ID to a script, wouldn't it break the graph by placing 7/21 in the second row (rather than the last)?

All I was trying to say (albeit confusingly) is to make sure that you handle the case where a spreadsheet ID is provided. Currently the code doesn't distinguish between the two because it inserts a new row 2 regardless if it's a new or an existing spreadsheet.

The way master works does result in a reverse chronological order, but personally I don't see that as a problem (I'm sure others could): most of the time I find myself interested in the latest results and having to scroll all the way to the bottom is more work. This is also why the individual sheets have the row of the totals at the top of the sheet and not at the bottom, as usual.

But I should be clear that I'm fine with whatever order you want to use.

**Comment by miketaylr** [comment]
Ah, makes sense. No real need to rock the boat -- we can tweak a graph to expect ascending vs descending order (or whichever makes sense).

In case we decide we want to append, here's the patch I dropped from #94:

```diff
diff --git a/spreadsheet.js b/spreadsheet.js
index 175b3e8..71954d9 100644
--- a/spreadsheet.js
+++ b/spreadsheet.js
@@ -84,29 +84,6 @@ async function updateSummary(sheets, spreadsheetId, date) {
             continue;
         }
         sheetId = properties.sheetId;
-        const requests = [];
-        const range = {
-            "sheetId": sheetId,
-            "startRowIndex": 1,
-            "endRowIndex": 2,
-            "startColumnIndex": 0,
-            "endColumnIndex": 19,
-        };
-
-        // Insert new row.
-        requests.push({
-            "insertRange": {
-                range,
-                "shiftDimension": 'ROWS',
-            },
-        });
-        const batchUpdateRequest = { requests };
-
-        result = await sheets.spreadsheets.batchUpdate({
-            spreadsheetId,
-            resource: batchUpdateRequest,
-        });
-
         const summary = [
             `=DATEVALUE("${title}")`,
             `='${title}'!$C$1`,
@@ -128,13 +105,15 @@ async function updateSummary(sheets, spreadsheetId, date) {
             `='${title}'!$U$1`,
             `='${title}'!$V$1`,
         ];
-        result = await sheets.spreadsheets.values.update({
+
+        result = await sheets.spreadsheets.values.append({
             spreadsheetId,
             range: `Summary!A2:S2`,
             resource: {
                 values: [summary],
             },
             valueInputOption,
+            insertDataOption: "INSERT_ROWS",
         });
 
         // format the first column as a date
```
